tests/nonbonded/test_nonbonded.py

```diff
@@ -0,0 +1,276 @@
+from dataclasses import replace
+from typing import Optional
+
+import numpy as np
+import pytest
+from common import GradientTest, gen_nonbonded_params_with_4d_offsets, prepare_system_params, prepare_water_system
+from numpy.typing import NDArray
+
+from timemachine import potentials
+from timemachine.ff import Forcefield
+from timemachine.ff.handlers import openmm_deserializer
+from timemachine.md import builders
+from timemachine.testsystems.dhfr import setup_dhfr
+
+np.set_printoptions(linewidth=500)
+
+pytestmark = [pytest.mark.memcheck]
+
+
+@pytest.fixture(scope="module")
+def _dhfr_system():
+    # This test checks hilbert curve re-ordering gives identical results
+    host_fns, _, host_coords, box = setup_dhfr()
+
+    nonbonded_bp = next(bp for bp in host_fns if isinstance(bp.potential, potentials.Nonbonded))
+    host_conf = host_coords
+    nonbonded_bp.potential.cutoff = 1.2
+    return nonbonded_bp, host_conf, box
+
+
+@pytest.fixture(scope="function")
+def dhfr_system(_dhfr_system):
+    nonbonded_bp, host_conf, box = _dhfr_system
+    return nonbonded_bp, np.array(host_conf), np.array(box)
+
+
+@pytest.mark.parametrize("precision", [np.float64, np.float32])
+def test_nblist_hilbert(dhfr_system, precision):
+    """
+    This test makes sure that hilbert ordering has no impact on numerical results. The
+    computed forces, energies, etc. should be bitwise identical.
+    """
+
+    np.random.seed(2021)
```

Comment:
Note that since this is now in `tests/nonbonded`, the autouse fixture `set_random_seed` applies and initially calls `np.random.seed(2022)` (I'm not sure the autouse fixture was a good idea...)

Reply:
For now I just left it as is, should still be safe this way if I am not misunderstanding the code.